src/components/protocol_handler/test/protocol_handler_tm_test.cc

```diff
@@ -192,17 +202,27 @@ class ProtocolHandlerImplTest : public ::testing::Test {
     // use protection OFF
     const bool callback_protection_flag = PROTECTION_OFF;
 #endif  // ENABLE_SECURITY
+    std::vector<std::string> empty;
```

Comment:
I have found an issue with any tests using this function (from what I can tell)

I wrote one optimization where it checks whether any rejected params exist (!rejected_params.empty()), but for some reason even when it should have been empty it wasn't considered to be so. This caused my code to fail because the list would always appear used.

I tried printing out the first value and size of this vector in these case and this is what I got:

```
[ RUN      ] ProtocolHandlerImplTest.EndSession_SessionObserverReject
rejected_params size: 18446726482061310441
rejected_params: (null)
unknown file: Failure

Unexpected mock function call - returning default value.
    Function call: SendMessageToDevice(1)
          Returns: 0
Google Mock tried the following 2 expectations, but none matched:

/home/jacob/sdl_core/src/components/protocol_handler/test/protocol_handler_tm_test.cc:231: tried expectation #0: EXPECT_CALL(transport_manager_mock, SendMessageToDevice(ControlMessage(FRAME_DATA_START_SERVICE_ACK, PROTECTION_OFF)))...
  Expected arg #0: Unprotected control message 
           Actual: 1, Message with data 0x3, not 0x2
         Expected: to be called once
           Actual: never called - unsatisfied and active
/home/jacob/sdl_core/src/components/protocol_handler/test/protocol_handler_tm_test.cc:761: tried expectation #1: EXPECT_CALL(transport_manager_mock, SendMessageToDevice( ControlMessage(FRAME_DATA_END_SERVICE_NACK, PROTECTION_OFF)))...
  Expected arg #0: Unprotected control message 
           Actual: 1, Message with data 0x3, not 0x6
         Expected: to be called once
           Actual: never called - unsatisfied and active
/home/jacob/sdl_core/src/components/protocol_handler/test/protocol_handler_tm_test.cc:768: Failure
Value of: waiter->WaitFor(times, kAsyncExpectationsTimeout)
  Actual: false
Expected: true
/home/jacob/sdl_core/src/components/protocol_handler/test/protocol_handler_tm_test.cc:231: Failure
Actual function call count doesn't match EXPECT_CALL(transport_manager_mock, SendMessageToDevice(ControlMessage(FRAME_DATA_START_SERVICE_ACK, PROTECTION_OFF)))...
         Expected: to be called once
           Actual: never called - unsatisfied and active
[  FAILED  ] ProtocolHandlerImplTest.EndSession_SessionObserverReject (10003 ms)
```

Obviously, the size is wildly incorrect (18446726482061310441), which leads me to believe that for some reason  the empty vector is not being initialized correctly.